### PR TITLE
AL-27: Using shared HMPPS kotlin library for building web client

### PR DESCRIPTION
**Pull request by iterative-it**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies {
   implementation("org.springframework.boot:spring-boot-starter-oauth2-client")
   implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
   implementation("org.springframework.boot:spring-boot-starter-data-jpa")
+  implementation("uk.gov.justice.service.hmpps:hmpps-kotlin-spring-boot-starter:0.2.1")
```
Comment (mikehalmamoj) [inline build.gradle.kts:19]: Don't need these - they are added by the new library:
```
  implementation("org.springframework.boot:spring-boot-starter-security")
  implementation("org.springframework.boot:spring-boot-starter-oauth2-client")
  implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
```

File: helm_deploy/values-preprod.yaml [hunk 2]
```diff
@@ -9,9 +9,9 @@ generic-service:
 
   env:
     APPLICATIONINSIGHTS_CONFIGURATION_FILE: applicationinsights.dev.json
-    HMPPS_AUTH_URL: https://sign-in-preprod.hmpps.service.justice.gov.uk/auth
-    USER_MANAGEMENT_API_URL: https://manage-users-api-preprod.hmpps.service.justice.gov.uk
-    PRISONER_SEARCH_API_URL: https://prisoner-search-preprod.prison.service.justice.gov.uk
+    API_BASE_URL_HMPPS_AUTH: "https://sign-in-dev.hmpps.service.justice.gov.uk/auth"
```
Comment (mikehalmamoj) [inline helm_deploy/values-preprod.yaml:12]: not preprod?

Comment (iterative-it) [reply helm_deploy/values-preprod.yaml]: Thanks! Copy paste error ;)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/health/HealthCheckTest.kt [hunk 3]
```diff
@@ -32,14 +59,6 @@ class HealthCheckTest : IntegrationTestBase() {
       )
   }
 
-  @Test
-  fun `Health page reports database ok`() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/health/HealthCheckTest.kt:36]: Seems like a good test - wondering why it's removed.

Comment (iterative-it) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/health/HealthCheckTest.kt:36]: I moved it to be part of the overall health OK test. I did wonder about having an OK test for each dependency but went with following the pattern from the template PR

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/health/HealthCheckTest.kt (hunk 3 above)
Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/integration/health/HealthCheckTest.kt:36]: Nah that seems fine

**Pull request merged**